Lib/ctypes/_util.py

```diff
@@ -0,0 +1,15 @@
+from sys import maxsize
+
+def _last_version(libnames, sep):
+    def _num_version(libname):
+        # "libxyz.so.MAJOR.MINOR" => [MAJOR, MINOR]
+        parts = libname.split(sep)
+        nums = []
+        try:
+            while parts:
+                nums.insert(0, int(parts.pop()))
```

Comment:
I would prefer to iterate in the reverse order rather than modifying the list:

for part in reversed(parts):
   try: ...
   except ValueError: break